src/CoreWCF.Primitives/src/CoreWCF/Security/SamlSerializer.cs

```diff
@@ -1,9 +1,115 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Security.Cryptography.X509Certificates;
+using System.Xml;
+using CoreWCF.IdentityModel.Selectors;
+using CoreWCF.IdentityModel.Tokens;
+using MSSAmlSerializer = Microsoft.IdentityModel.Tokens.Saml.SamlSerializer;
+using Microsoft.IdentityModel.Tokens.Saml;
+
 namespace CoreWCF.Security
 {
     public class SamlSerializer
     {
+        private readonly MSSAmlSerializer _mSSamlSerializer;
+        public SamlSerializer()
+        {
```

Comment:
This class appears on a lot of public api surface, but it's not usable for anything. Can you make the ReadToken method public virtual again so that someone can derive the class and replace the implementation.